datadog/detect.go

```diff
@@ -26,6 +26,23 @@ func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error
 		return libcnb.DetectResult{Pass: false}, nil
 	}
 
+	// If both BP_DATADOG_ENABLED and BP_NATIVE_IMAGE are enabled, don't require jvm-application plan and prepare for native-image only
```

Comment:
I'm wondering if this is strictly necessary. The `native-image` buildpack will require `jvm-application` (see [here](https://github.com/paketo-buildpacks/native-image/blob/main/native/detect.go#L61)), so it should be provided during a native-image build and OK to have as a requirement here also.

Or were you seeing specific issues without this change?

Reply:
Here is what I tried:
```bash
pack build <appname> \
    --buildpack <fullpath-to-datadog-buildpack-repo>/datadog \
    --buildpack paketo-buildpacks/java-native-image \
    --builder paketobuildpacks/builder:tiny \
    --env BP_NATIVE_IMAGE=true \
    --env BP_DATADOG_ENABLED=true
```
And here is the result:
```
===> DETECTING
ERROR: No buildpack groups passed detection.
ERROR: Please check that you are running against the correct path.
ERROR: failed to detect: no buildpacks participating
ERROR: failed to build: executing lifecycle: failed with status code: 20
```
So I understand `jvm-application` is missing when `datadog` buildpack is run first.
And I had to run it before `native-image` buildpack to ensure it set its compiler argument (as opposed to the _non-native-image_ build, where `datadog` buildpack can run after application compilation and only set run arguments).

Reply:
OK, so you have to split out the composite buildpack to insert Datadog into the right place then.

```
pack build <appname> \
    --buildpack paketo-buildpacks/bellsoft-liberica \
    --buildpack paketo-buildpacks/syft \
    --buildpack paketo-buildpacks/maven \  # or gradle
    --buildpack <fullpath-to-datadog-buildpack-repo>/datadog \
    --buildpack paketo-buildpacks/native-image \
    --buildpack paketo-buildpacks/spring-boot \
    --builder paketobuildpacks/builder:tiny \
    --env BP_NATIVE_IMAGE=true \
    --env BP_DATADOG_ENABLED=true
```

I'm guessing that position would be OK as it happens before native-image, but you could move it before Maven/Gradle too if necessary. Does that work? Or do you still get detection failures?

Reply:
Thanks! By splitting the composite buildpack, it works well for the two cases:
 
1. After Maven/Gradle:

```bash
pack build <appname> \
    --buildpack paketo-buildpacks/bellsoft-liberica \
    --buildpack paketo-buildpacks/syft \
    --buildpack paketo-buildpacks/maven \  # or gradle
    --buildpack paketo-buildpacks/executable-jar \
    --buildpack <fullpath-to-datadog-buildpack-repo>/datadog \
    --buildpack paketo-buildpacks/native-image \
    --buildpack paketo-buildpacks/spring-boot \
    --builder paketobuildpacks/builder:tiny \
    --env BP_NATIVE_IMAGE=true \
    --env BP_DATADOG_ENABLED=true
```

3. Before Maven/Grade:

```bash
pack build <appname> \
    --buildpack paketo-buildpacks/bellsoft-liberica \
    --buildpack paketo-buildpacks/syft \
    --buildpack <fullpath-to-datadog-buildpack-repo>/datadog \
    --buildpack paketo-buildpacks/maven \  # or gradle
    --buildpack paketo-buildpacks/executable-jar \
    --buildpack paketo-buildpacks/native-image \
    --buildpack paketo-buildpacks/spring-boot \
    --builder paketobuildpacks/builder:tiny \
    --env BP_NATIVE_IMAGE=true \
    --env BP_DATADOG_ENABLED=true
```

Please note I tested with Maven only (it should be the same for Gradle right?) and I had to add the `paketo-buildpacks/executable-jar` buildpack too).

With those last tests, is there anything more I can do to get the PR ready?

Reply:
You're right. I forgot to add executable-jar to the list. That is expected. 

Gradle should work the same as Maven. I'll run some more tests though to confirm.

Overall, code is looking good. I need to run some more tests, but I think we're getting close. I hope to get the testing done this week.

I will say, once we merge this into Datadog. We'll need another PR to add Datadog to the Java Native Image composite buildpack. That'll be what actually makes it usable for end users, at least without needing all of the `--buildpack` arguments.

Reply:
That was just to make the path clear. It's not necessary to open now.

Reply:
Did you see the test error btw? It looks minor, but if you can rebase and push up a fix that would be helpful.

Reply:
> if you can rebase and push up a fix that would be helpful

Done ✔️

Reply:
@cforce Yes, I think that should work but it's going to duplicate some of the buildpacks. I think those should just no-op, but if you see issues then break up java-native-image into the subset of component buildpacks you need it to run. Like https://github.com/paketo-buildpacks/datadog/pull/149#discussion_r1126169572

Reply:
The `<buildpacks>` block is an exact list of buildpacks to run. Some buildpacks like `java-native-image` are "composite" buildpacks, or buildpacks of buildpacks. The are a convenience so you can quickly specify a set of buildpacks. Those just get expanded before they are run, but they get expanded in the order defined in that buildpacks, which makes it tricky to use them when you need to inject a buildpack into the middle. That's why I typically don't use them when I need total control over the order buildpacks will run.